tests/test_bs.py

```diff
@@ -0,0 +1,229 @@
+        match = re.search(r"Local node identity >> (?P<bootstarap_key>\w+)", bootstrap_pod_logs)
+        bs_info.bs_key = match.group('bootstarap_key')
+        return bs_info
+
+    def fin():
+        global bs_info
+        delete_deployment(bs_info.bs_deployment_name, testconfig['namespace'])
+
+    request.addfinalizer(fin)
+    return _setup_bootstrap_in_namespace(testconfig['namespace'])
+
+
+@pytest.fixture
+def setup_clients(request, setup_bootstrap):
+
+    def _setup_clients_in_namespace(name_space):
+        global bs_info, client_info
+        client_info = NodeInfo()
+        resp = create_deployment(CLIENT_DEPLOYMENT_FILE, name_space,
+                                 bootstrap_ip=bs_info.bs_pod_ip,
+                                 bootstrap_port='7513',
+                                 bootstrap_key=bs_info.bs_key)
+        client_info.bs_deployment_name = resp.metadata._name
+        namespaced_pods = client.CoreV1Api().list_namespaced_pod(namespace=name_space, include_uninitialized=True).items
+        client_pods = list(filter(lambda i: i.metadata.name.startswith(client_info.bs_deployment_name), namespaced_pods))
+
+        print("Number of client pods: {0}".format(len(client_pods)))
+        for c in client_pods:
+            while True:
+                resp = client.CoreV1Api().read_namespaced_pod(name=c.metadata.name, namespace=name_space)
+                if resp.status.phase != 'Pending':
+                    break
+                time.sleep(1)
+        return client_pods
+
+    def fin():
+        global client_info
+        delete_deployment(client_info.bs_deployment_name, testconfig['namespace'])
+
+    request.addfinalizer(fin)
+    return _setup_clients_in_namespace(testconfig['namespace'])
+
+
+#==============================================================================
+#    TESTS
+#==============================================================================
+
+current_index = 'kubernetes_cluster-2019.02.2*'
+
+
+def test_bootstrap(load_config, setup_bootstrap):
+    time.sleep(10)
+    assert setup_bootstrap.bs_key == query_bootstrap_es(current_index,
+                                                        testconfig['namespace'],
+                                                        setup_bootstrap.bs_pod_name)
+
+
+def test_client(load_config, setup_clients):
+    time.sleep(10)
+    global client_info
+    peers = query_es_client_bootstrap(current_index, testconfig['namespace'], client_info.bs_deployment_name)
+    assert len(setup_clients) == peers
+
+
+def test_gossip(load_config, setup_clients):
+    global client_info
+
+    # send message to client via rpc
+    client_ip = setup_clients[0].status.pod_ip
+    print("Sending gossip from client ip: {0}".format(client_ip))
+
+    api = 'v1/broadcast'
+    data = '{"data":"foo"}'
+    p = subprocess.Popen(['./kubectl-cmd.sh', '%s' % client_ip, "%s" % data, api], stdin=subprocess.PIPE, stdout=subprocess.PIPE,
```

Comment:
you should probably wrap the RPC call logic in a function.

Reply:
ok